{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: 'genomic_analyses//SEQ//S_SEQ//SARS2correct_Sequences.fasta'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[1;32m~\\AppData\\Local\\Temp/ipykernel_17692/2242955366.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      6\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      7\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 8\u001b[1;33m \u001b[1;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'genomic_analyses//SEQ//'\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mseq\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m+\u001b[0m \u001b[1;34m'_SEQ//'\u001b[0m\u001b[1;33m+\u001b[0m\u001b[0mstr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhcov\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m+\u001b[0m\u001b[1;34m'correct_Sequences.fasta'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mfile\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      9\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mseq_rec\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfile\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'fasta'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m         \u001b[0mseq_id\u001b[0m\u001b[1;33m=\u001b[0m \u001b[0mseq_rec\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mdescription\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: 'genomic_analyses//SEQ//S_SEQ//SARS2correct_Sequences.fasta'"
     ]
    }
   ],
   "source": [
    "df=pd.DataFrame()\n",
    "\n",
    "hcov= 'SARS2' #choose name\n",
    "seq='S'  #full or S\n",
    "\n",
    "\n",
    "\n",
    "with open('genomic_analyses//SEQ//'+str(seq)+ '_SEQ//'+str(hcov)+'correct_Sequences.fasta') as file:\n",
    "    for seq_rec in SeqIO.parse(file, 'fasta'):\n",
    "        seq_id= seq_rec.description\n",
    "        if seq_id:\n",
    "            record= {'id': seq_id, 'seq': str(seq_rec.seq.lower())}\n",
    "            df=df.append(record, ignore_index=True)\n",
    "print(len(df), df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[4, 3, 19, 18, 27, 26, 30, 29, 34, 33, 35, 34, 36, 35, 48, 47, 52, 51, 55, 54, 56, 55, 59, 58, 61, 60, 62, 61, 63, 62, 64, 63, 65, 64, 69, 68, 71, 70, 78, 77, 81, 80, 101, 100, 110, 109, 115, 114, 118, 117, 119, 118, 122, 121, 126, 125, 127, 126, 130, 129, 133, 132, 134, 133, 141, 140, 142, 141, 154, 153, 156, 155, 158, 157, 162, 161, 164, 163, 165, 164, 184, 183, 188, 187, 194, 193, 196, 195, 198, 197, 202, 201, 204, 203, 205, 204, 206, 205, 208, 207, 210, 209, 214, 213, 229, 228, 231, 230, 234, 233, 235, 234, 237, 236, 238, 237, 251, 250, 258, 257, 259, 258, 262, 261, 267, 266, 268, 267, 270, 269, 271, 270, 272, 271, 275, 274, 276, 275, 280, 279, 283, 282, 288, 287, 290, 289, 293, 292, 302, 301, 308, 307, 310, 309, 311, 310, 313, 312, 314, 313, 320, 319, 321, 320, 322, 321, 323, 322, 327, 326, 328, 327, 329, 328, 331, 330, 332, 331, 335, 334, 341, 340, 342, 341, 343, 342, 348, 347, 350, 349, 354, 353, 356, 355, 357, 356, 359, 358, 361, 360, 362, 361, 364, 363, 366, 365, 367, 366, 368, 367, 369, 368, 370, 369, 371, 370, 372, 371, 373, 372, 378, 377, 383, 382, 384, 383, 398, 397, 399, 398, 400, 399, 401, 400, 409, 408, 413, 412, 418, 417, 419, 418, 426, 425, 427, 426, 428, 427, 437, 436, 447, 446, 449, 448, 455, 454, 456, 455, 457, 456, 470, 469, 475, 474, 479, 478, 485, 484, 488, 487, 496, 495, 497, 496, 502, 501, 503, 502, 505, 504, 512, 511, 519, 518, 520, 519, 522, 521, 523, 522, 524, 523, 526, 525, 544, 543, 546, 545, 576, 575, 580, 579, 581, 580, 586, 585, 598, 597, 599, 598, 604, 603, 609, 608, 613, 612, 616, 615, 620, 619, 623, 622, 625, 624, 626, 625, 627, 626, 642, 641, 644, 643, 647, 646, 648, 647, 649, 648, 650, 649, 651, 650, 659, 658, 662, 661, 664, 663, 671, 670, 672, 671, 678, 677, 679, 678, 681, 680, 682, 681, 688, 687, 694, 693, 702, 701, 703, 702, 705, 704, 715, 714, 722, 721, 734, 733, 736, 735, 745, 744, 748, 747, 751, 750, 756, 755, 767, 766, 769, 768, 774, 773, 776, 775, 780, 779, 783, 782, 784, 783, 786, 785, 787, 786, 796, 795, 800, 799, 807, 806, 816, 815, 817, 816, 818, 817, 819, 818, 832, 831, 833, 832, 834, 833, 836, 835, 848, 847, 858, 857, 860, 859, 865, 864, 867, 866, 872, 871, 873, 872, 874, 873, 875, 874, 877, 876, 878, 877, 879, 878, 883, 882, 897, 896, 903, 902, 907, 906, 908, 907, 909, 908, 928, 927, 931, 930, 932, 931, 934, 933, 935, 934, 940, 939, 942, 941, 947, 946, 948, 947, 949, 948, 951, 950, 952, 951, 953, 952, 956, 955, 957, 956, 959, 958, 960, 959, 964, 963, 965, 964, 968, 967, 971, 970, 977, 976, 978, 977, 979, 978, 980, 979, 981, 980, 982, 981, 983, 982, 984, 983, 985, 984, 992, 991, 995, 994, 996, 995, 997, 996, 998, 997, 1002, 1001, 1003, 1002, 1004, 1003, 1005, 1004, 1007, 1006, 1008, 1007, 1009, 1008, 1011, 1010, 1012, 1011, 1013, 1012, 1014, 1013, 1016, 1015, 1017, 1016, 1018, 1017, 1019, 1018, 1020, 1019, 1021, 1020, 1022, 1021, 1023, 1022, 1026, 1025, 1027, 1026, 1028, 1027, 1029, 1028, 1031, 1030, 1032, 1031, 1035, 1034, 1036, 1035, 1039, 1038, 1040, 1039, 1041, 1040, 1045, 1044, 1051, 1050, 1054, 1053, 1056, 1055, 1057, 1056, 1058, 1057, 1059, 1058, 1060, 1059, 1061, 1060, 1062, 1061, 1063, 1062, 1064, 1063, 1065, 1064, 1068, 1067, 1072, 1071, 1073, 1072, 1078, 1077, 1084, 1083, 1086, 1085, 1087, 1086, 1088, 1087, 1089, 1088, 1090, 1089, 1092, 1091, 1093, 1092, 1096, 1095, 1097, 1096, 1099, 1098, 1104, 1103, 1106, 1105, 1107, 1106, 1111, 1110, 1112, 1111, 1113, 1112, 1117, 1116, 1118, 1117, 1119, 1118, 1121, 1120, 1122, 1121, 1124, 1123, 1125, 1124, 1126, 1125, 1127, 1126, 1128, 1127, 1131, 1130, 1132, 1131, 1135, 1134, 1136, 1135, 1137, 1136, 1139, 1138, 1141, 1140, 1142, 1141, 1145, 1144, 1147, 1146, 1149, 1148, 1150, 1149, 1154, 1153, 1155, 1154, 1160, 1159, 1162, 1161, 1164, 1163, 1165, 1164, 1166, 1165, 1168, 1167, 1169, 1168, 1170, 1169, 1172, 1171, 1173, 1172, 1174, 1173, 1176, 1175, 1177, 1176, 1178, 1177, 1179, 1178, 1180, 1179, 1183, 1182, 1184, 1183, 1185, 1184, 1186, 1185, 1189, 1188, 1190, 1189, 1194, 1193, 1196, 1195, 1197, 1196, 1198, 1197, 1199, 1198, 1203, 1202, 1209, 1208, 1211, 1210, 1212, 1211, 1213, 1212, 1215, 1214, 1221, 1220, 1228, 1227, 1229, 1228, 1230, 1229, 1232, 1231, 1233, 1232, 1234, 1233, 1237, 1236, 1238, 1237, 1242, 1241, 1246, 1245, 1247, 1246, 1248, 1247, 1249, 1248, 1250, 1249, 1254, 1253, 1255, 1254, 1256, 1255, 1257, 1256, 1259, 1258, 1260, 1259, 1261, 1260, 1262, 1261, 1263, 1262, 1264, 1263, 1267, 1266, 1268, 1267, 1269, 1268, 1273, 1272, 1278, 1277, 1280, 1279, 1281, 1280, 1282, 1281, 1285, 1284, 1286, 1285, 1287, 1286, 1288, 1287, 1289, 1288, 1290, 1289, 1294, 1293, 1295, 1294, 1303, 1302, 1308, 1307, 1310, 1309, 1311, 1310, 1312, 1311, 1314, 1313, 1315, 1314, 1319, 1318, 1320, 1319, 1329, 1328, 1332, 1331, 1334, 1333, 1335, 1334, 1339, 1338, 1340, 1339, 1341, 1340, 1342, 1341, 1347, 1346, 1348, 1347, 1349, 1348, 1359, 1358, 1360, 1359, 1361, 1360, 1362, 1361, 1368, 1367, 1373, 1372, 1376, 1375, 1377, 1376, 1378, 1377, 1379, 1378, 1381, 1380, 1385, 1384, 1389, 1388, 1392, 1391, 1393, 1392, 1394, 1393, 1395, 1394, 1397, 1396, 1399, 1398, 1400, 1399, 1401, 1400, 1402, 1401, 1405, 1404, 1408, 1407, 1411, 1410, 1412, 1411, 1417, 1416, 1421, 1420, 1424, 1423, 1425, 1424, 1427, 1426, 1431, 1430, 1441, 1440, 1442, 1441, 1443, 1442, 1444, 1443, 1445, 1444, 1447, 1446, 1448, 1447, 1449, 1448, 1450, 1449, 1451, 1450, 1452, 1451, 1453, 1452, 1454, 1453, 1455, 1454, 1456, 1455, 1457, 1456, 1458, 1457, 1460, 1459, 1461, 1460, 1462, 1461, 1463, 1462, 1464, 1463, 1465, 1464, 1466, 1465, 1467, 1466, 1469, 1468, 1470, 1469, 1471, 1470, 1477, 1476, 1481, 1480, 1486, 1485, 1489, 1488, 1491, 1490, 1493, 1492, 1494, 1493, 1495, 1494, 1497, 1496, 1498, 1497, 1499, 1498, 1503, 1502, 1505, 1504, 1507, 1506, 1508, 1507, 1509, 1508, 1510, 1509, 1511, 1510, 1512, 1511, 1516, 1515, 1518, 1517, 1520, 1519, 1522, 1521, 1524, 1523, 1527, 1526, 1528, 1527, 1530, 1529, 1532, 1531, 1533, 1532, 1536, 1535, 1537, 1536, 1538, 1537, 1539, 1538, 1540, 1539, 1541, 1540, 1546, 1545, 1547, 1546, 1548, 1547, 1549, 1548, 1550, 1549, 1551, 1550, 1552, 1551, 1553, 1552, 1555, 1554, 1556, 1555, 1557, 1556, 1558, 1557, 1559, 1558, 1565, 1564, 1567, 1566, 1568, 1567, 1569, 1568, 1570, 1569, 1571, 1570, 1573, 1572, 1574, 1573, 1575, 1574, 1582, 1581, 1583, 1582, 1586, 1585, 1588, 1587, 1589, 1588, 1593, 1592, 1594, 1593, 1595, 1594, 1604, 1603, 1605, 1604, 1624, 1623, 1625, 1624, 1626, 1625, 1627, 1626, 1628, 1627, 1629, 1628, 1630, 1629, 1631, 1630, 1632, 1631, 1633, 1632, 1634, 1633, 1640, 1639, 1650, 1649, 1661, 1660, 1662, 1661, 1664, 1663, 1665, 1664, 1666, 1665, 1667, 1666, 1674, 1673, 1677, 1676, 1678, 1677, 1679, 1678, 1680, 1679, 1681, 1680, 1682, 1681, 1685, 1684, 1686, 1685, 1688, 1687, 1689, 1688, 1692, 1691, 1693, 1692, 1696, 1695, 1702, 1701, 1704, 1703, 1705, 1704, 1707, 1706, 1714, 1713, 1716, 1715, 1719, 1718, 1720, 1719, 1721, 1720, 1722, 1721, 1723, 1722, 1725, 1724, 1726, 1725, 1728, 1727, 1733, 1732, 1735, 1734, 1736, 1735, 1737, 1736, 1739, 1738, 1740, 1739, 1741, 1740, 1742, 1741, 1743, 1742, 1750, 1749, 1751, 1750, 1752, 1751, 1754, 1753, 1759, 1758, 1760, 1759, 1761, 1760, 1764, 1763, 1766, 1765, 1767, 1766, 1776, 1775, 1777, 1776, 1778, 1777, 1779, 1778, 1780, 1779, 1789, 1788, 1790, 1789, 1791, 1790, 1792, 1791, 1794, 1793, 1802, 1801, 1807, 1806, 1811, 1810, 1814, 1813, 1816, 1815, 1821, 1820, 1822, 1821, 1823, 1822, 1825, 1824, 1828, 1827, 1836, 1835, 1837, 1836, 1839, 1838, 1841, 1840, 1846, 1845, 1847, 1846, 1851, 1850, 1852, 1851, 1853, 1852, 1856, 1855, 1857, 1856, 1859, 1858, 1860, 1859, 1861, 1860, 1864, 1863, 1866, 1865, 1870, 1869, 1877, 1876, 1879, 1878, 1881, 1880, 1882, 1881, 1884, 1883, 1885, 1884, 1889, 1888, 1894, 1893, 1895, 1894, 1896, 1895, 1897, 1896, 1898, 1897, 1899, 1898, 1900, 1899, 1901, 1900, 1903, 1902, 1904, 1903, 1905, 1904, 1906, 1905, 1907, 1906, 1909, 1908, 1910, 1909, 1911, 1910, 1912, 1911, 1913, 1912, 1914, 1913, 1915, 1914, 1940, 1939, 1941, 1940, 1942, 1941, 1943, 1942, 1944, 1943, 1945, 1944, 1948, 1947, 1949, 1948, 1952, 1951, 1953, 1952, 1956, 1955, 1957, 1956, 1966, 1965, 1967, 1966, 1973, 1972, 1974, 1973, 1980, 1979, 1981, 1980, 1982, 1981, 1984, 1983, 1989, 1988, 1992, 1991, 1994, 1993, 1995, 1994, 1999, 1998, 2001, 2000, 2004, 2003, 2005, 2004, 2006, 2005, 2007, 2006, 2008, 2007, 2009, 2008, 2013, 2012, 2017, 2016, 2018, 2017, 2019, 2018, 2020, 2019, 2021, 2020, 2022, 2021, 2029, 2028, 2030, 2029, 2036, 2035, 2037, 2036, 2039, 2038, 2044, 2043, 2045, 2044, 2047, 2046, 2054, 2053, 2056, 2055, 2057, 2056, 2060, 2059, 2061, 2060, 2065, 2064, 2066, 2065, 2068, 2067, 2069, 2068, 2070, 2069, 2074, 2073, 2077, 2076, 2080, 2079, 2081, 2080, 2082, 2081, 2083, 2082, 2084, 2083, 2089, 2088, 2090, 2089, 2093, 2092, 2094, 2093, 2095, 2094, 2096, 2095, 2097, 2096, 2098, 2097]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>seq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>hCoV-19/Brazil/RJ-FIOCRUZ-20761/2020</td>\n",
       "      <td>---------------------------------------ctttcga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>hCoV-19/Bahamas/33398/2021</td>\n",
       "      <td>----------------------------------------tttcga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>hCoV-19/USA/AZ-ASU25023/2021</td>\n",
       "      <td>----------------------------------------------...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>hCoV-19/Argentina/INEI100606/2021</td>\n",
       "      <td>----------------------------------------------...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>hCoV-19/Slovakia/CeMM3325/2020</td>\n",
       "      <td>----------------------------------------------...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>941</th>\n",
       "      <td>hCoV-19/Slovakia/BA_22_00029745/2022</td>\n",
       "      <td>---aaaggtttataccttcccaggtaacaaaccaaccaactttcga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>942</th>\n",
       "      <td>hCoV-19/Spain/MD-HGUGM-6507189/2022</td>\n",
       "      <td>-------------------------------ccaaccaactttcga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>943</th>\n",
       "      <td>hCoV-19/Singapore/4932/2022</td>\n",
       "      <td>------------taccttcccaggtaacaaaccaaccaacttttga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>944</th>\n",
       "      <td>hCoV-19/Singapore/4968/2022</td>\n",
       "      <td>-----aggtttataccttcccrggtaacaaaccaaccaactttyga...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>945</th>\n",
       "      <td>hCoV-19/Canada/MB-CPL-421038/2022</td>\n",
       "      <td>----------------------------------------------...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>946 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       id  \\\n",
       "0    hCoV-19/Brazil/RJ-FIOCRUZ-20761/2020   \n",
       "1              hCoV-19/Bahamas/33398/2021   \n",
       "2            hCoV-19/USA/AZ-ASU25023/2021   \n",
       "3       hCoV-19/Argentina/INEI100606/2021   \n",
       "4          hCoV-19/Slovakia/CeMM3325/2020   \n",
       "..                                    ...   \n",
       "941  hCoV-19/Slovakia/BA_22_00029745/2022   \n",
       "942   hCoV-19/Spain/MD-HGUGM-6507189/2022   \n",
       "943           hCoV-19/Singapore/4932/2022   \n",
       "944           hCoV-19/Singapore/4968/2022   \n",
       "945     hCoV-19/Canada/MB-CPL-421038/2022   \n",
       "\n",
       "                                                   seq  \n",
       "0    ---------------------------------------ctttcga...  \n",
       "1    ----------------------------------------tttcga...  \n",
       "2    ----------------------------------------------...  \n",
       "3    ----------------------------------------------...  \n",
       "4    ----------------------------------------------...  \n",
       "..                                                 ...  \n",
       "941  ---aaaggtttataccttcccaggtaacaaaccaaccaactttcga...  \n",
       "942  -------------------------------ccaaccaactttcga...  \n",
       "943  ------------taccttcccaggtaacaaaccaaccaacttttga...  \n",
       "944  -----aggtttataccttcccrggtaacaaaccaaccaactttyga...  \n",
       "945  ----------------------------------------------...  \n",
       "\n",
       "[946 rows x 2 columns]"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# take out all sequences that have more than 20% unknown  --- only full sequences\n",
    "ambiguous=[]\n",
    "for i, row in df.iterrows():\n",
    "    #for all rows that are not the reference\n",
    "    # look for any nt that are not atgc or a gap (-)\n",
    "    re.sub(r\"[^atgc-]\", 'n', row['seq'])\n",
    "    # calculate the percentage of nucleotides that are either n or gap\n",
    "    calc=len(re.findall( r\"[n]\", str(row['seq'])))/ len(row['seq'])\n",
    "        # if more that 1 % is gap , make list to drop seq and its ref\n",
    "    if calc > 0.01:\n",
    "        ambiguous.append(i)\n",
    "        ambiguous.append(i-1)\n",
    "print(ambiguous)      \n",
    "df= df.drop(ambiguous).reset_index(drop=True)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "def dataframetofasta(dataframe, fastaname):\n",
    "        #create fasta file\n",
    "        file= fastaname + \".fasta\"\n",
    "        f= open(file, 'a')\n",
    "        #write to the file\n",
    "        for i, row in dataframe.iterrows():\n",
    "            lines='>'+ row['id']+'\\n'+ row['seq'] + \"\\n\"\n",
    "            f.write(lines)\n",
    "        f.close()\n",
    "\n",
    "\n",
    "# !!!!\n",
    "# to do: change name and make code run everything at once\n",
    "dataframetofasta(df, 'genomic_analyses//SEQ//'+str(seq)+ '_SEQ//notambiguous//'+str(hcov)+'_notambiguouscorrect')\n"
   ]
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "9c9ae641c5235c2f1f03348eb2e8dc8801508e47084873ad2b25c816a6e3304f"
  },
  "kernelspec": {
   "display_name": "Python 3.9.13 64-bit (windows store)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.13"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
